README.md

```diff
@@ -127,6 +128,28 @@ actual   : have
 If either value is longer than one line both are displayed in *multi-line* format.
 
 
+### `assert_not_equal`
+
+Fail if the two parameters, actual and unexpected value respectively, are equal.
+
+```bash
+@test 'assert_not_equal()' {
+  assert_not_equal 'foobar' 'foobar'
+}
+```
+
+On failure, the expected and actual values are displayed.
+
+```
+-- values should not be equal --
+unexpected : foobar
+actual     : foobar
```

Comment:
Do we really need the same value to be printed twice?

Reply:
@martin-schulze-vireso, I think it's a matter of opinion. `assertpy` does reports this way:
```
def test_dummy_test() -> None:
>       assert_that(1 + 1).is_not_equal_to(2)
E       AssertionError: Expected <2> to be not equal to <2>, but was.
```
while `assertj` does it this way:
```
java.lang.AssertionError: 
Expecting actual:
  2
not to be equal to:
  2
```

While I'm not advocating that _bats-assert_ should be "just like everybody else", I do think that it should feel somewhat familiar, as well as follow the other patterns in the existing code base.

Reply:
I guess regardless of language the question is: does the comparison equal output string comparison or can the string values differ for something that is considered equal?

Reply:
> does the comparison equal output string comparison or can the string values differ for something that is considered equal?


If I'm understanding the question correctly, `assert_not_equal` only throws if the two values are exactly equal (as defined by `[[ "$1" == "$2" ]]`).

Reply:
Yes, I can't see how there will be a difference between the comparison and the string values. However, bash has always been good for surprises in weird corner cases. We can leave the code as is. I was just thinking about lengthy texts being printed twice.

Reply:
Sounds good.

Speaking of corner cases, I just discovered an infinite loop if you run the following:
```
run assert_equal "" ""
assert_test_fail
```

I have to dig some more before reporting it, but gotta love surprises.